src/Internal/Declaration/WorkflowInstance.php

```diff
@@ -62,15 +82,28 @@ public function __construct(
             $this->signalQueue->attach($method, $this->signalHandlers[$method]);
         }
 
+        $updateValidators = $prototype->getValidateUpdateHandler();
+        foreach ($prototype->getUpdateHandlers() as $method => $reflection) {
+            $fn = $this->createHandler($reflection);
+            $this->updateHandlers[$method] = fn(UpdateInput $input): mixed => ($this->updateExecutor)($input, $fn);
+            // Register validate update handlers
+            $this->validateUpdateHandlers[$method] = \array_key_exists($method, $updateValidators)
+                ? fn(UpdateInput $input): mixed => ($this->updateValidator)(
+                    $input,
+                    $this->createHandler($updateValidators[$method]),
+                )
+                : null;
```

Comment:
Ideally, throw right away at workflow registration time (or however early you can).

It could, but, I'm not inclined to allow passing multiple handlers to the attribute. I'd say let people just call a shared function from two places.